### PR TITLE
Add ab04md

### PR DESCRIPTION
**Pull request by KybernetikJo**
Add ab04md

TODO:
- [x] add docstring
- [x] add unit tests
 - [x] check behavior: intent(in, out, copy, ...), see #200 

```python
import slycot
import numpy as np

A = np.array([[1.0, 0.5],
              [0.5, 1.0]])
B = np.array([[0.0, -1.0],
              [1.0, 0.0]])
C = np.array([[-1.0, 0.0],
              [0.0, 1.0]])
D = np.array([[1.0, 0.0],
              [0.0, -1.0]])

Ad = np.array([[-1.0, -4.0],
               [-4.0, -1.0]])
Bd = np.array([[2.8284, 0.0],
               [0.0, -2.8284]])
Cd = np.array([[0.0, 2.8284],
               [-2.8284, 0.0]])
Dd = np.array([[-1.0, 0.0],
               [0.0, -3.0]])

n = 2
m = 2
p = 2

Ad_t, Bd_t, Cd_t, Dd_t = slycot.ab04md('C',n,m,p,A,B,C,D)

print("Ad_t = \n", Ad_t)
print("Bd_t = \n", Bd_t)
print("Cd_t = \n", Cd_t)
print("Dd_t = \n", Dd_t)

Ac_t, Bc_t, Cc_t, Dc_t = slycot.ab04md('D',n,m,p,Ad_t,Bd_t,Cd_t,Dd_t)

print("Ac_t = \n", Ac_t)
print("Bc_t = \n", Bc_t)
print("Cc_t = \n", Cc_t)
print("Dc_t = \n", Dc_t)

print(np.allclose(A,Ac_t))
print(np.allclose(B,Bc_t))
print(np.allclose(C,Cc_t))
print(np.allclose(D,Dc_t))
```

output
```
Ad_t = 
 [[-1. -4.]
 [-4. -1.]]
Bd_t = 
 [[-2.82842712 -0.        ]
 [-0.          2.82842712]]
Cd_t = 
 [[-0.          2.82842712]
 [-2.82842712  0.        ]]
Dd_t = 
 [[3. 0.]
 [0. 1.]]
Ac_t = 
 [[1.  0.5]
 [0.5 1. ]]
Bc_t = 
 [[ 0. -1.]
 [ 1. -0.]]
Cc_t = 
 [[-1.  0.]
 [-0.  1.]]
Dc_t = 
 [[ 1.  0.]
 [ 0. -1.]]
True
True
True
True
```

### CODE REVIEW

**Comment by KybernetikJo** [comment]
I have added the docstring, 2 unittests and discussed f2py behavior in #200. I would also keep here f2py intent(in,out,copy) for the state space matrices A,B,C,D.

Remark:
There is quite a bit of variation in the style of docstrings and unittests in slycot.

File: slycot/analysis.py [hunk 1]
```diff
@@ -139,6 +139,79 @@ def ab01nd(n, m, A, B, jobz='N', tol=0, ldwork=None):
...
+    """ At,Bt,Ct,Dt = ab04md(type_bn, n, m, p, A, B, C, D, [alpha, beta,ldwork])
+
+    Parameters
+    ----------
+    type_t : {'D','C'}
+            Indicates the type of the original system and the
+            transformation to be performed as follows:
+            = 'D':  discrete-time   -> continuous-time;
+            = 'C':  continuous-time -> discrete-time.
+    n : input int
```
Comment (bnavigator) [inline slycot/analysis.py:152]: Please adhere to numpydoc #100 
```suggestion
    n : int
```

File: slycot/__init__.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
     # import slycot.examples
 
     # Analysis routines (15/40 wrapped)
-    from .analysis import ab01nd, ab05md, ab05nd, ab07nd, ab08nd, ab08nz
+    from .analysis import ab01nd, ab04md, ab05md, ab05nd, ab07nd, ab08nd, ab08nz
```
Comment (bnavigator) [inline slycot/__init__.py:16]: Please increase the count from above.

A reformat of the whle ab block seems in order, too

**Comment by bnavigator** [comment]
Sorry for not being responsive as usual. I made a few remarks about formal things.

**Comment by KybernetikJo** [comment]
I have incorporated the suggestions and tried to improve the docstrings.
However, the .pfy, .py files and the docstring are very different in slycot in general and specifically in the analysis module.

I think we would need templates to unify the docstrings of all routines to meet the numpydoc specification.
Currently, only a few routines like ab01nd meet the numpydoc specification.

File: slycot/__init__.py (hunk 2 above)
Comment (bnavigator) [reply slycot/__init__.py:16]: #205

File: slycot/analysis.py [hunk 3]
```diff
@@ -139,6 +139,79 @@ def ab01nd(n, m, A, B, jobz='N', tol=0, ldwork=None):
         Z = None
     return Ac, Bc, ncont, indcon, nblk, Z, tau
 
+def ab04md(type_t, n, m, p, A, B, C, D, alpha=1.0, beta=1.0, ldwork=None):
+    """ At,Bt,Ct,Dt = ab04md(type_bn, n, m, p, A, B, C, D, [alpha, beta,ldwork])
```
Comment (bnavigator) [inline slycot/analysis.py:143]:
```suggestion
    """ At,Bt,Ct,Dt = ab04md(type_t, n, m, p, A, B, C, D, [alpha, beta, ldwork])
```

File: slycot/analysis.py [hunk 4]
```diff
@@ -139,6 +139,79 @@ def ab01nd(n, m, A, B, jobz='N', tol=0, ldwork=None):
...
+    B : (n,m) ndarray
+        The leading n-by-m part of this array must contain the system input
+        matrix B.
+    C : (p,n) ndarray
+        The leading p-by-n part of this array must contain the system output
+        matrix C.
+    D : (p,m) ndarray
+        The leading p-by-m part of this array must contain the system direct
+        transmission matrix D.
+    alpha : double
```
Comment (bnavigator) [inline slycot/analysis.py:174]:
```suggestion
    alpha : double, optional
```

File: slycot/tests/test_ab04md.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+import unittest
+from slycot import analysis
+import numpy as np
+
+from scipy.linalg import eig
+from numpy.testing import assert_equal, assert_allclose
+
+
+class test_ab04md(unittest.TestCase):
```
Comment (bnavigator) [inline slycot/tests/test_ab04md.py:12]: `unittest.TestCase` not needed

**Pull request merged**